libnd4j/include/ops/declarable/helpers/cpu/image_resize.cpp

```diff
@@ -350,12 +426,443 @@ namespace helpers {
+                                            0, 1, yWai, y_ptr_0, y_ptr_1, y_ptr_2, y_ptr_3, xWai);
+                                    cached_value_2[0] = computeYInterpolation(
+                                            0, 2, yWai, y_ptr_0, y_ptr_1, y_ptr_2, y_ptr_3, xWai);
+
+                                case 1:
+                                    cached_value_0[1] = computeYInterpolation(
+                                            1, 0, yWai, y_ptr_0, y_ptr_1, y_ptr_2, y_ptr_3, xWai);
+                                    cached_value_1[1] = computeYInterpolation(
+                                            1, 1, yWai, y_ptr_0, y_ptr_1, y_ptr_2, y_ptr_3, xWai);
+                                    cached_value_2[1] = computeYInterpolation(
+                                            1, 2, yWai, y_ptr_0, y_ptr_1, y_ptr_2, y_ptr_3, xWai);
+
+                                case 2:
+                                    cached_value_0[2] = computeYInterpolation(
+                                            2, 0, yWai, y_ptr_0, y_ptr_1, y_ptr_2, y_ptr_3, xWai);
+                                    cached_value_1[2] = computeYInterpolation(
+                                            2, 1, yWai, y_ptr_0, y_ptr_1, y_ptr_2, y_ptr_3, xWai);
+                                    cached_value_2[2] = computeYInterpolation(
+                                            2, 2, yWai, y_ptr_0, y_ptr_1, y_ptr_2, y_ptr_3, xWai);
+
+                                case 3:
+                                    cached_value_0[3] = computeYInterpolation(
+                                            3, 0, yWai, y_ptr_0, y_ptr_1, y_ptr_2, y_ptr_3, xWai);
+                                    cached_value_1[3] = computeYInterpolation(
+                                            3, 1, yWai, y_ptr_0, y_ptr_1, y_ptr_2, y_ptr_3, xWai);
+                                    cached_value_2[3] = computeYInterpolation(
+                                            3, 2, yWai, y_ptr_0, y_ptr_1, y_ptr_2, y_ptr_3, xWai);
+                                    break;
+                            }
+                            pOutput[x * numChannels + 0] =
+                                    compute(cached_value_0, xWai._weight0, xWai._weight1,
+                                            xWai._weight2, xWai._weight3);
+                            pOutput[x * numChannels + 1] =
+                                    compute(cached_value_1, xWai._weight0, xWai._weight1,
+                                            xWai._weight2, xWai._weight3);
+                            pOutput[x * numChannels + 2] =
+                                    compute(cached_value_2, xWai._weight0, xWai._weight1,
+                                            xWai._weight2, xWai._weight3);
+                        }
+                    } else {
+                        for (Nd4jLong x = 0; x < resizerState.outWidth; ++x) {
```

Comment:
Are you sure this loop should not be parallelized? Or this is internal loop?

Reply:
this is internal loop